bin/python_scripts/find_invalid_tags.py

```diff
@@ -44,20 +44,50 @@ def get_all_tags():
     cursor = connection.cursor()
     sql = """
     SELECT tag.name, count(tag.name) FROM tag, package, package_tag
-    WHERE tag.id = package_tag.tag_id AND package.id = package_tag.package_id AND package.state = 'active'
+    WHERE tag.id = package_tag.tag_id AND package.id = package_tag.package_id 
+    AND package.state = 'active' AND package_tag.state = 'active'
     GROUP BY tag.name ORDER BY count(tag.name) DESC;
     """
 
     cursor.execute(sql)
     return cursor
 
 
+def get_tag_data():
+    cursor = connection.cursor()
+    sql = """
+    SELECT tag.name, package_tag.package_id, package_tag.tag_id, tag.vocabulary_id 
+    FROM tag, package, package_tag
+    WHERE tag.id = package_tag.tag_id AND package.id = package_tag.package_id 
+    AND package.state = 'active' AND package_tag.state = 'active'
+    ORDER BY tag.name;
+    """
+
+    cursor.execute(sql)
+    return cursor
+
+
 def create_tag_list(rows):
     with io.open("invalid_tags.txt", "w+", encoding="utf-8") as f:
         f.writelines(rows)
     logger.info('TXT - Written %s lines to invalid_tags.txt', len(rows) - 1)
 
 
+def create_invalid_tag_data(invalid_tags):
+    rows = []
+    rows.append(u"index, tag, package_id, tag_id, vocabulary_id\n")
+    for i, tag in enumerate(get_tag_data()):
+        tag_name, package_id, tag_id, vocabulary_id = tag
+        if tag_name in invalid_tags:
+            rows.append(u"{}, {}, {}, {}, {}".format(
+                str(i + 1), tag_name, package_id, tag_id, vocabulary_id) + '\n')
+
+    with io.open("invalid_tags_data.csv", "w+", encoding="utf-8") as f:
+        f.writelines(rows)
```

Comment:
It might be nicer to use the [`csv` module](https://docs.python.org/2/library/csv.html) here:

```python
with io.open("invalid_tags_data.csv", "w+", encoding="utf-8") as file:
    writer = csv.writer(file)
    writer.writerow(['index', 'tag', 'package_id', 'tag_id', 'vocabulary_id'])
    for i, tag in enumerate(get_tag_data()):
        ...
        writer.writerow([str(i + 1), tag_name, package_id, tag_id, vocabulary_id])
```

Reply:
Does it give us much benefit over just using the standard IO file writer?

Reply:
I would argue it makes the code easier to read and nicer dealing with row structures rather than formatted strings. Probably one of the main benefits is that it'll properly support records that have got `,` characters in them.

Reply:
ok, thats a good enough benefit